core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -943,7 +952,7 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeOptionalString(routing);
             out.writeOptionalString(parent);
             out.writeLong(version);
-            
+
             out.writeByte(versionType.getValue());
             out.writeLong(autoGeneratedIdTimestamp);
             out.writeVLong(seqNo);
```

Comment:
Assertions are indeed not enabled on the standalone integration tests. I think they should be, they would have caught at least one bug in core (that I found after I enabled assertions there), and the issue here before changing the serialization to `(read|write)Long`. I will open a PR soon.